src/ensembl/src/content/app/browser/browserHelper.ts

```diff
@@ -14,3 +23,77 @@ export function getChrLocationStr(
 
   return `${chrCode}:${startBp}-${endBp}`;
 }
+
+export const getRegionValidationResult = (
+  validationInfo: RegionValidationResponse | null
+) => {
+  let genomeIdError,
+    regionParamError,
+    parseError,
+    regionError,
+    startError,
+    endError,
+    regionId;
+
+  if (validationInfo) {
+    if (validationInfo.message) {
+      genomeIdError = validationInfo.message.genome_id;
+      regionParamError = validationInfo.message.region;
+    } else {
+      if (!validationInfo.is_parseable) {
+        parseError = RegionErrors.PARSE_ERROR;
+      }
+      if (validationInfo.region && !validationInfo.region.is_valid) {
+        regionError = validationInfo.region.error_message;
+      }
+      if (validationInfo.start && !validationInfo.start.is_valid) {
+        startError = validationInfo.start.error_message;
+      }
+      if (validationInfo.end && !validationInfo.end.is_valid) {
+        endError = validationInfo.end.error_message;
+      }
+      if (validationInfo.region_id) {
+        regionId = validationInfo.region_id;
+      }
+    }
+  }
+
+  // this is till parse error is properly returned in the response
+  regionError = parseError || regionError;
+
+  return {
+    errorMessages: {
+      genomeIdError,
+      regionParamError,
+      regionError,
+      startError,
+      endError
+    },
+    regionId
+  };
+};
+
+export const validateRegion = async (params: {
+  regionInput: string;
+  genomeId: string | null;
+  onSuccess: (regionId: string) => void;
+  onError: (errorMessageList: any) => void;
```

Comment:
any -> JSONValue (from types/JSON)?

Reply:
I added a specific type for this with a new commit